web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -325,114 +395,131 @@ export function Navigation() {
   if (hideNav) {
     return null;
   }
-
   return (
-    <Box
+    <Container
       as="nav"
       onMouseLeave={() => resetExpandedSection()}
-      onKeyUp={e => e.key === 'Escape' && resetExpandedSection()}
+      onKeyUp={e => e.key === 'Escape' && resetExpandedSection(false)}
       onBlur={(event: React.FocusEvent<HTMLDivElement, Element>) => {
         if (!event.currentTarget.contains(event.relatedTarget)) {
           resetExpandedSection();
         }
       }}
-      css={`
-        position: relative;
-        width: var(--sidenav-width);
-        z-index: ${zIndexMap.sideNavContainer};
-      `}
+      className={
+        stickyMode &&
+        currentPageSection &&
+        !!debouncedSection &&
+        !debouncedSection.standalone
+          ? 'sticky-mode'
+          : ''
+      }
     >
       <SideNavContainer>
         <PanelBackground />
         <SearchSection
-          navigationSections={[...navSections, topMenuSection]}
+          navigationSections={[...combinedSideNavSections, topMenuSection]}
```

Comment:
`[...combinedSideNavSections, topMenuSection]` will always return a new array, so anything depending on this will always re-render. As you've memo'd `combinedSideNavSections`, should you memo this?